### PR TITLE
Align test & lint

### LINKED ISSUE #67: Code coverage error
```
Traceback (most recent call last):
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/cmdline.py", line 746, in do_run
    runner.run()
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/execfile.py", line 247, in run
    exec(code, main_mod.__dict__)
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/pytest/__main__.py", line 5, in <module>
    raise SystemExit(pytest.console_main())
SystemExit: ExitCode.OK

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/pysetup/.venv/bin/coverage", line 8, in <module>
    sys.exit(main())
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/cmdline.py", line 871, in main
    status = CoverageScript().command_line(argv)
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/cmdline.py", line 588, in command_line
    return self.do_run(options, args)
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/cmdline.py", line 753, in do_run
    self.coverage.save()
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/control.py", line 659, in save
    data = self.get_data()
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/control.py", line 727, in get_data
    if self._collector and self._collector.flush_data():
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/collector.py", line 442, in flush_data
    self.covdata.add_lines(self.mapped_file_dict(self.data))
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/sqldata.py", line 438, in add_lines
    self._choose_lines_or_arcs(lines=True)
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/sqldata.py", line 495, in _choose_lines_or_arcs
    with self._connect() as con:
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/sqldata.py", line 300, in _connect
    self._create_db()
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/sqldata.py", line 249, in _create_db
    with db:
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/sqldata.py", line 1037, in __enter__
    self._connect()
  File "/opt/pysetup/.venv/lib/python3.9/site-packages/coverage/sqldata.py", line 1019, in _connect
    self.con = sqlite3.connect(filename, check_same_thread=False)
sqlite3.OperationalError: unable to open database file
ERROR: 1
make: *** [test] Error 1
```

### LINKED ISSUE #79: `make lint` is too slow
`make lint  3.54s user 1.93s system 10% cpu 51.279 total`

Shouldn't we run `pre-commit run --all-files` instead ? (`98.66s user 5.41s system 522% cpu 19.920 total`)

### PR DESCRIPTION
**Pull request by grahamalama**
Chatted with @leplatrem about #81, and I added some more changes on top of that PR here.

The idea with this PR is that `infra/lint.sh` and `infra/test.sh` are the "source of truth" for how we want to run linting and tests respectively. We then use these scripts in precommit, make, Github Actions, and wherever else we want to run tests and linting.

For Github Actions, we also now run linting and tests directly in the action runner, rather than running through Docker. This resulted in a speed-up of the checks.

Fixes #79 
Fixes #67

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,32 +26,30 @@ help:
...
 lint:
-	docker-compose -f ./docker-compose.yaml -f ./tests/infra/docker-compose.lint.yaml build \
-		--build-arg userid=${_UID} --build-arg groupid=${_GID} lint
-
+	docker-compose run --rm web infra/lint.sh
 
 .PHONY: shell
 shell:
-	docker-compose -f ./docker-compose.yaml run web
+	docker-compose -f ./docker-compose.yaml run web /bin/sh
```
Comment (leplatrem) [inline Makefile:43]: is `-f ./docker-compose.yaml ` superfluous here?

Comment (grahamalama) [reply Makefile]: https://github.com/mozilla/jira-bugzilla-integration/pull/88/commits/9d71842bd7ff8699ea0b152575c7cc7b63513146

**Comment by grahamalama** [comment]
Thanks for the approval @bsieber-mozilla. @leplatrem, looking for your review as well since I built on top of your initial PR.

File: infra/lint.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  case $subcommand in
+    "black" | "isort")
+      case $1 in
+        "--fix")
+          check=""
+        ;;
+      esac
+      case $subcommand in
+        "isort") isort;;
+        "black") black;;
```
Comment (leplatrem) [inline infra/lint.sh:75]: You don't pass the `$check` argument here ?

**Review by leplatrem — APPROVED** [review]
Great 🎉 

nit: the Github jobs are called `lint-build` and `test-build`, which was relevant when we were testing containers. Now they can be rename lint and test IMO

**Review by leplatrem — CHANGES_REQUESTED** [review]
I just realized we exclude tests from certain checks.

When including `tests` in `isort` we face the issue I was having, where `black` unwraps and `isort` wraps back.

File: infra/lint.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+black () {
+  $POETRY_RUN black ${check:+--check} src tests
+}
+detect_secrets () {
+  # Scan only files fixed into the repo, omit poetry.lock
+  SECRETS_TO_SCAN=`git ls-tree --full-tree -r --name-only HEAD | grep -v poetry.lock`
+  $POETRY_RUN detect-secrets-hook $SECRETS_TO_SCAN --baseline .secrets.baseline
+}
+isort () {
+  $POETRY_RUN isort ${check:+--check-only} src
```
[the remark this answers is not in the document]
Comment (grahamalama) [reply infra/lint.sh]: I decided that if we're doing `src` and `tests`, we might as well do it all https://github.com/mozilla/jira-bugzilla-integration/pull/88/commits/c272ef2f579429f8883849c341ffa2193af66ad2 .

`$POETRY_RUN isort ${check:+--check-only} .`

Also, the `--recursive` flag is deprecated in isort 5. The `.` [should be enough](https://pycqa.github.io/isort/index.html#using-isort) to recursively search all files.

**Comment by grahamalama** [comment]
> When including tests in isort we face the issue I was having, where black unwraps and isort wraps back

Something I just had to do that fixed this:
- make sure you have isort `^5.10.1` installed in your virtualenv
- make sure you don't have an old `.isort.cfg` file laying around -- I noticed I still had one in my directory, and I think that was the thing that was causing problems since that will take precedence over pyproject.toml if it's present.

I think when we do that, isort will respect the "black" profile in `pyproject.toml`

File: infra/lint.sh [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+}
+pylint () {
+  $POETRY_RUN pylint src tests
+}
+mypy () {
+  $POETRY_RUN mypy src
```
[the remark this answers is not in the document]
Comment (grahamalama) [reply infra/lint.sh:25]: Do we want/need to run typechecking on tests?

**Review by leplatrem — APPROVED** [review]

File: infra/lint.sh (hunk 2 above)
Comment (grahamalama) [reply infra/lint.sh:75]: I think the `$check` param argument will be picked up when it's set above. I guess it acts like a global?

Tbh, I'm not the best Bash author, so I'd welcome a PR to make this bash script nicer. I was also thinking of rewriting this in Python -- we don't _need_ to torture ourselves with Bash.

**Pull request merged**